common-rest/src/test/java/org/uniprot/api/rest/service/query/UniProtQueryProcessorTest.java

```diff
@@ -33,19 +37,53 @@
 class UniProtQueryProcessorTest {
     private static final String FIELD_NAME = "acc";
     private UniProtQueryProcessor processor;
+    private UniProtQueryProcessorConfig config;
 
     @BeforeEach
     void setUp() {
         Map<String, String> whitelistFields = new HashMap<>();
         whitelistFields.put("go", "^[0-9]+$");
+        config = UniProtQueryProcessorConfig.builder()
+                .optimisableFields(
+                        singletonList(
+                                searchFieldWithValidRegex(FIELD_NAME, "^P[0-9]+$")))
+                .whiteListFields(whitelistFields)
+                .build();
         processor =
                 UniProtQueryProcessor.newInstance(
-                        UniProtQueryProcessorConfig.builder()
-                                .optimisableFields(
-                                        singletonList(
-                                                searchFieldWithValidRegex(FIELD_NAME, "^P[0-9]+$")))
-                                .whiteListFields(whitelistFields)
-                                .build());
+                        config);
+    }
+
+    @Test
+    void manyQueriesToProcess() {
+        List<String> queries =
+                List.of(
+                        "(*)",
+                        "this is a default query",
+                        "(Methanococcoides burtonii Hel308)",
+                        "(ethylsterigmatocystin oxidoreductase)",
+                        "(L2EFL_DROME)",
+                        "(L8I5P5)",
+                        "(another type) AND (length:[10 TO 20])",
+                        "Serine\\/alpha is okay");
+
+        ExecutorService executorService = Executors.newFixedThreadPool(10);
+        for (int i = 0; i < 10000; i++) {
+            executorService.submit(
+                    () ->
+                            UniProtQueryProcessor.newInstance(config).processQuery(
+                                    queries.get((int) (Math.random() * queries.size()))));
+        }
+
+
+        executorService.shutdown();
+        try {
+            if (!executorService.awaitTermination(800, TimeUnit.MILLISECONDS)) {
+                executorService.shutdownNow();
+            }
+        } catch (InterruptedException e) {
+            executorService.shutdownNow();
+        }
     }
 
```

Comment:
What does it test? Should we assert some expected result in the test case?

Reply:
totally agree! i forgot about this -- and just added a proper test

Reply:
cool. please merge.